### PR TITLE
Feature/add configurations using node config module

### PR DESCRIPTION
**Pull request by jianwel**

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "REDIS_URL" : "redis://:redis_pass@redis_db:6379/1",
+    "LOCAL_SECRET_KEY" : "0x0000000000000000000000000000000000000000000000000000000000000001",
+    "EVENT_CHUNK_SIZE" : 50, 
+    "LOGGING_LEVEL" : "info",
+    "CLOUDWATCH_LEVEL" : "info",
+    "ELASTICSEARCH_LEVEL" : "info",
+    "RPC_SERVER_PORT" : 2000,
+    "CONTRACT_FIXTURES_URL" : "https://s3.amazonaws.com/shipchain-contracts/meta.json",
+    "DATABASE_URL" : "psql://engine:engine@psql:5432/engine",
+    "isDeployedStage" : false,
```
Comment (ajhodges) [inline config/default.json]: I think the casing of all options should be consistent, and since they are constants, I think the typical style guide says ALL_CAPS format is the way to go

File: config/local_example.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (ajhodges) [inline config/local_example.txt]: What is the purpose of the local_example.txt file? Is a local config necessary or is it covered by default.json?

I think this should be moved to local.json, or local.json.example, and it should be valid json. If you want to annoate with comments, keep the comments in a js style block comment at the top of the file (even though this is not valid json, I think it is an okay compromise).

File: config/stage.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "EncryptorContainer" : {
+    "defaultEncryptor" : "AwsPrivateKeyDBFieldEncryption"
+    },
+    "isDeployedStage" : true, 
+    "rdsAwsSecretKey" : "ENGINE_RDS_STAGE",
```
Comment (ajhodges) [inline config/stage.json]: rdsAwsSecretKey and masterPrivateKeySecretName, etc, are always ENGINE_RDS_{ENV}, etc. So these I think we can determine programmatically based on ENV instead of setting in each properties file.

File: package.json [hunk 4]
```diff
@@ -57,7 +58,7 @@
     "lint": "eslint 'src/**/*.ts' 'rpc/**/*.ts' ",
     "lint-junit": "eslint 'src/**/*.ts' 'rpc/**/*.ts' --format junit -o reports/junit/lint-junit.xml ",
     "lint-html": "eslint  'src/**/*.ts' 'rpc/**/*.ts' --format html  -o reports/lint.html ",
-    "test": "jest --detectOpenHandles --forceExit --runInBand",
+    "test": "jest --detectOpenHandles --forceExit --runInBand --verbose=false",
```
Comment (ajhodges) [inline package.json]: What is the reason for this change? If necessary for this PR, can you please make a note about it in the PR description

File: config/default.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    "isGettingBestGasStationPriceNeeded" : false,
+    "isDeployingLocalContracts" : true
```
Comment (ajhodges) [inline config/default.json]: DEPLOY_CONTRACTS is a better name I think

File: config/default.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    "LOCAL_SECRET_KEY" : "0x0000000000000000000000000000000000000000000000000000000000000001",
+    "EVENT_CHUNK_SIZE" : 50, 
+    "LOGGING_LEVEL" : "info",
+    "CLOUDWATCH_LEVEL" : "info",
+    "ELASTICSEARCH_LEVEL" : "info",
+    "RPC_SERVER_PORT" : 2000,
+    "CONTRACT_FIXTURES_URL" : "https://s3.amazonaws.com/shipchain-contracts/meta.json",
+    "DATABASE_URL" : "psql://engine:engine@psql:5432/engine",
+    "isDeployedStage" : false,
+    "isGettingBestGasStationPriceNeeded" : false,
```
Comment (ajhodges) [inline config/default.json]: 'GPO_ETH_GAS_STATION'

File: package.json (hunk 4 above)
Comment (jianwel) [reply package.json]: oh, this is for printing the log when running the tests, forgot to remove it...
